src/postgres/inventory/collection/PgCollection.ts

```diff
@@ -133,23 +132,35 @@ class PgCollection implements Collection<PgClassType.Value> {
       async (values: Array<PgClassType.Value>): Promise<Array<PgClassType.Value>> => {
         const insertionIdentifier = Symbol()
 
+        // Get the fields that actually have been used in any of the values, so
+        // we can leave the rest out of the query completely to prevent
+        // permission issues
+        const fields = new Map()
+        values.forEach((map) => {
+          map.forEach((value, key) => {
+            if (typeof value !== 'undefined') {
+              fields.set(key, this.type.fields.get(key))
+            }
```

Comment:
I hadn't really considered that yet, but I can see how that could become an issue. I unfortunately do not see a way to avoid that with this approach.

Another approach to this problem would be to detect which columns can be inserted on through introspection instead. This only has to be done at startup/watch updates, and does not involve extra looping, so this would effectively be `O(1)`. We would go back to sometimes specifying `default` for columns that we could leave out completely, but as long as you have the rights to set those columns this is fine. I am not (yet) familiar enough with the project to achieve this though, unfortunately.